src/ruleset_builder.cpp

```diff
@@ -36,6 +36,10 @@ namespace {
 std::set<rule *> target_to_rules(const std::vector<parser::rule_target_spec> &targets,
     const std::unordered_map<std::string_view, rule::ptr> &rules, const rule_tag_map &rules_by_tags)
 {
+    if (rules_by_tags.empty()) {
+        return {};
+    }
```

Comment:
Does this make sense? The `else` branch doesn't use `rules_by_tags`. Or is it that tags are mandatory and so there are no rules if the map is empty? It could be clearer.

Reply:
I'm not sure why I added this, rules should always have at least type and category so I'll just remove it.